onnx/defs/traditionalml/defs.cc

```diff
@@ -314,103 +314,147 @@ static const char* LabelEncoder_ver2_doc = R"DOC(
     "Sally"] would be mapped to [-1, 5, 5, 6, 6].<br>
     Since this operator is an one-to-one mapping, its input and output shapes
     are the same. Notice that only one of 'keys_*'/'values_*' can be set.<br>
+    When 'keys_as_tensor' and 'values_as_tensor' are used, the default value
+    must be specified in 'default_as_tensor' attribute as a singleton with the
+    same type as 'values_as_tensor'. Additionally, 'keys_as_tensor' and
+    'values_as_tensor' must be one dimensional and have the same length.<br>
     For key look-up, bit-wise comparison is used so even a float NaN can be
     mapped to a value in 'values_*' attribute.<br>
 )DOC";
 
 ONNX_ML_OPERATOR_SET_SCHEMA(
     LabelEncoder,
-    2,
+    4,
     OpSchema()
-        .SetDoc(LabelEncoder_ver2_doc)
+        .SetDoc(LabelEncoder_ver4_doc)
         .Input(0, "X", "Input data. It can be either tensor or scalar.", "T1")
         .Output(0, "Y", "Output data.", "T2")
         .TypeConstraint(
             "T1",
-            {"tensor(string)", "tensor(int64)", "tensor(float)"},
+            {"tensor(string)", "tensor(int64)", "tensor(float)", "tensor(int32)", "tensor(int16)"},
             "The input type is a tensor of any shape.")
         .TypeConstraint(
             "T2",
-            {"tensor(string)", "tensor(int64)", "tensor(float)"},
+            {"tensor(string)", "tensor(int64)", "tensor(float)", "tensor(int32)", "tensor(int16)"},
             "Output type is determined by the specified 'values_*' attribute.")
         .Attr(
-            "keys_strings",
-            "A list of strings. One and only one of 'keys_*'s should be set.",
-            AttributeProto::STRINGS,
+            "keys_as_tensor",
```

Comment:
minor nit: may be `keys_tensor` and `values_tensor` and `default_tensor` would be closer to the existing naming pattern.

Reply:
Why not preserve the existing naming pattern, and use "keys_tensor" etc. as the attribute name?